pkg/operator/statefulset.go

```diff
@@ -75,13 +75,38 @@ func CreateConfigReloader(
 		Limits:   v1.ResourceList{},
 		Requests: v1.ResourceList{},
 	}
-	if config.CPU != "0" {
-		resources.Limits[v1.ResourceCPU] = resource.MustParse(config.CPU)
-		resources.Requests[v1.ResourceCPU] = resource.MustParse(config.CPU)
+
+	// Prioritize specific flags over generic ones
+	// If config.CPU has been set but not config.CPURequest/config.CPULimit, default "" to config.CPU
+	// If config.Memory has been set but not config.MemoryRequest or config.MemoryLimit, default "" to config.Memory
+	if config.CPU != "" {
+		if config.CPURequest == "" {
+			config.CPURequest = config.CPU
+		}
+		if config.CPULimit == "" {
+			config.CPULimit = config.CPU
+		}
+	}
+	if config.Memory != "" {
+		if config.MemoryRequest == "" {
+			config.MemoryRequest = config.Memory
+		}
+		if config.MemoryLimit == "" {
+			config.MemoryLimit = config.Memory
+		}
+	}
+
+	if config.CPURequest != "0" {
```

Comment:
Should this not go under the `if config.CPU != ""` case? same for memory below?

Reply:
Currently, no memory/CPU limits are achieved in the code by not specifying a value for resources.Limits[v1.ResourceMemory] and resources.Limits[v1.ResourceCPU] respectively (as triggered by config.CPU and config.Memory being "0"). So this follows that by not specifying a value for the specific CPU/Memory request/limit if the value in the relevant config is "0".